src/daemon/rpmostree-sysroot-upgrader.cxx

```diff
@@ -419,6 +419,20 @@ rpmostree_sysroot_upgrader_pull_base (RpmOstreeSysrootUpgrader  *self,
 
   switch (refspec_type)
     {
+    case RPMOSTREE_REFSPEC_TYPE_CONTAINER:
+      {
+        if (override_commit)
+          return glnx_throw (error, "Specifying commit overrides for container:// is not supported");
+
+        char *repo_path = g_file_get_path (ostree_repo_get_path (self->repo));
+        auto commit = rpmostreecxx::import_container(std::string(repo_path), std::string(refspec));
```

Comment:
We can pass the `OstreeRepo` reference across to the Rust side instead of stringifying the path.  Or alternatively, pass the `OstreeSysroot` object which has the repo reference.  There are several examples of this, e.g.
`        fn daemon_sanitycheck_environment(sysroot: Pin<&mut OstreeSysroot>) -> Result<()>;` in `lib.rs`.